lib/rules/newline-before-return.js

```diff
@@ -101,10 +101,18 @@ module.exports = function(context) {
      * @private
      */
     function hasNewlineBefore(node) {
-        var tokenBefore = sourceCode.getTokenBefore(node);
-        var lineNumTokenBefore = tokenBefore.loc.end.line;
-        var lineNumNode = node.loc.start.line;
-        var commentLines = calcCommentLines(node, tokenBefore);
+        var tokenBefore = sourceCode.getTokenBefore(node),
+            lineNumNode = node.loc.start.line,
+            lineNumTokenBefore,
+            commentLines;
+
+        if (tokenBefore) {
+            lineNumTokenBefore = tokenBefore.loc.end.line;
+        } else {
+            lineNumTokenBefore = 0;     // assume "just before" start of file
```

Comment:
Agree with @kaicataldo that this comment is unclear without a lot of context. It would help to have a larger comment before the `if` block explaining that global `return` is a special case where we consider the previous line to be 0 so the rest of the functionality works correctly.